mathics/builtin/numbers/algebra.py

```diff
@@ -621,6 +621,11 @@ def powers_list(pf):
                         SymbolPlus, *[c[i] for c in contrib]
                     ).evaluate(evaluation)
                 return powers
+            else:
+                for i, pat in enumerate(var_pats):
+                    if match(pf, pat, evaluation):
+                        powers[i] = Integer(1)
```

Comment:
Integer1

Reply:
Personally I'd appreciate a comment here  explaining that we are using matching terms raised to the power 1 to make sure when the term is multiplied it isn't forgotten, and maybe a little example. 

A simple example reduced from  @Kalsofar's can be obtained by the zero parameter, e.g. 

```
Collect[ q[x] q[y] + q[[x],  q[x] ]
```